src/contour/TerminalSessionManager.cpp

```diff
@@ -84,49 +89,74 @@ TerminalSession* TerminalSessionManager::createSession()
     // sessions. This will work around it, by explicitly claiming ownership of the object.
     QQmlEngine::setObjectOwnership(session, QQmlEngine::CppOwnership);
 
-    // we can close application right after session has been created
-    _lastTabChange = std::chrono::steady_clock::now() - std::chrono::seconds(1);
-    _activeSession = session;
     return session;
 }
 
 void TerminalSessionManager::setSession(size_t index)
 {
     Require(index <= _sessions.size());
     managerLog()(std::format("SET SESSION: index: {}, _sessions.size(): {}", index, _sessions.size()));
+
     if (!isAllowedToChangeTabs())
         return;
 
-    Require(display != nullptr);
-    auto const pixels = display->pixelSize();
-    auto const totalPageSize = display->calculatePageSize() + _activeSession->terminal().statusLineHeight();
+    if (index < _sessions.size())
+        activateSession(_sessions[index]);
+    else
+        activateSession(createSessionInBackground());
+}
 
-    auto* oldSession = _activeSession;
+TerminalSession* TerminalSessionManager::activateSession(TerminalSession* session)
+{
+    managerLog()(
+        "Activating session ID {} at index {}", session->id(), getSessionIndexOf(session).value_or(-1));
 
-    if (index < _sessions.size())
+    if (_activeSession == session)
+    {
+        managerLog()("Session is already active. (index {}, ID {})", getCurrentSessionIndex(), session->id());
+        return session;
+    }
+
+    _previousActiveSession = _activeSession;
+    _activeSession = session;
+    _lastTabChange = std::chrono::steady_clock::now();
+    updateStatusLine();
+
+    if (display)
     {
-        _activeSession = _sessions[index];
+        managerLog()("Attaching display to session.");
+        auto const pixels = display->pixelSize();
+        auto const totalPageSize =
+            display->calculatePageSize() + _activeSession->terminal().statusLineHeight();
```

Comment:
```suggestion
            display->calculatePageSize() + _previousActiveSession->terminal().statusLineHeight();
```
to preserve the behavior

Reply:
Not sure why this one should be correct just yet :thinking: